rfcs/block-component.md

```diff
@@ -52,6 +52,59 @@ Props would be exposed for:
 
 > **Note** that we would likely _not_ use this component internally to build other baseui components. It's primarily a helper component for customers when building their apps.
 
+### `Block` API
+
+* `children: (props: Props) => React.Node` - Optional
+* `as: string | React.Node` - Optional
```

Comment:
Should this be React.ElementType instead of React.Node?

Reply:
From the [Flow & React docs](https://github.com/facebook/flow/blob/master/website/en/docs/react/types.md#reactnode-) it looks like it's the common approach for typing `children`.  It also looks like this is the common approach in the other BaseUI components.

I'm more familiar with TypeScript honestly, so maybe @nadiia or @gergelyke could provide some Flow insight here?

Reply:
Yeah, React.Node is usually what you want for `children` but I think what we have here is a different than children. `<div>` (or its transpiled version `React.createElement('div')`) is a React.Node, but styletron wouldn't be able to do anything with that. It just needs the string or the class, which it will pass to React.createElement internally.

Looks like styletron also uses ElementType internally:
https://github.com/styletron/styletron/blob/master/packages/styletron-react-core/src/index.js#L181

Reply:
🤦‍♂️ I was only looking at `children`. Just updated `as` to ElementType